src/uct/ib/dc/dc_mlx5_ep.c

```diff
@@ -613,7 +614,7 @@ ucs_status_t uct_dc_mlx5_ep_flush(uct_ep_h tl_ep, unsigned flags,
         return UCS_ERR_NO_RESOURCE;
     }
 
-    if (!uct_dc_mlx5_iface_dci_ep_can_send(ep)) {
+    if (!uct_dc_mlx5_iface_dci_ep_can_send(ep, flush_cancel)) {
```

Comment:
maybe better do the following instead of modifying `uct_dc_mlx5_iface_dci_ep_can_send`?
```
if (!uct_dc_mlx5_iface_dci_has_tx_resources(iface, ep->dci) && flush_cancel)
```

Reply:
but we need to check FC window and TX_WAIT flag set in case of flush(LOCAL)

Reply:
then can add branch here, like `else if (!uct_dc_mlx5_iface_dci_ep_can_send(ep))`

Reply:
done